Products/ZenRRD/runner.py

```diff
@@ -166,26 +170,28 @@ def processEnded(self, reason):
             self.stderr = "".join(self.stderr)
 
             msg = "Datasource: %s Received exit code: %s Output: \n%r"
-            data = [self.command.ds, self.exitCode, self.output]
+            data = [self.datasource.ds, self.exitCode, self.output]
             if self.stderr:
                 msg += "\nStandard Error:\n%r"
                 data.append(self.stderr)
             log.debug(msg, *data)
 
-        d, self.command_defer = self.command_defer, None
+        d, self._deferred = self._deferred, None
         if not d.called:
             d.callback(self)
 
 
+@implementer(IRunner)
 class SshRunner(object):
-    implements(IRunner)
+    """Execute commands on the remote end of an SSH connection."""
```

Comment:
This looked like an important change, implements() to @implementer(), what is the difference?

Reply:
They both accomplish the same thing.

`implements` is old-skool.  However, it implies "magic" because it somehow just knows what the class is.

`@implementer` is a class decorator and is the idiomatic way to apply changes to a class.  It also signals to anyone reading the code that something's being done to the class.  The `implements` function is just a function call and signals confusion to anyone unfamiliar with zope.interface APIs.